integration-tests/tests/preview-dataset.js

```diff
@@ -0,0 +1,74 @@
+const _ = require('lodash/fp')
+const { signIntoTerra, click, clickable, clickTableCell, findElement, findText, findTableCellText, getTableCellPath, getTableHeaderPath, input, waitForNoSpinners } = require('../utils/integration-utils')
+const { withWorkspace } = require('../utils/integration-helpers')
+const { withUserToken } = require('../utils/terra-sa-utils')
+const { dismissNotifications } = require('../utils/integration-utils')
+
+const datasetName = "Cell hashing with barcoded antibodies enables multiplexing and doublet detection for single cell genomics";
+
+const assertEquals = (expected, actual) => {
+    if (expected !== actual) {
+      throw new Error("Expected value " + expected + " does not equal actual " + actual)
+    }
+}
+
+const testPreviewDatasetFn = _.flow(
+  withWorkspace,
+  withUserToken
+)(async ({ testUrl, page, token, workspaceName }) => {
+  await page.goto(testUrl)
+  await waitForNoSpinners(page)
+
+  await findText(page, 'Browse Data')
+
+  await page.evaluate(() => window.configOverridesStore.set({ isDataBrowserVisible: true }))
+  await page.reload({ waitUntil: ['networkidle0', 'domcontentloaded'] })
+
+  await click(page, clickable({ textContains: 'Browse Data' }))
+  await signIntoTerra(page, token)
+  await dismissNotifications(page)
+
+  await click(page, clickable({ textContains: 'browse & explore' }))
+  await waitForNoSpinners(page)
+  await click(page, clickable({ textContains: 'Granted' }))
+  await click(page, clickable({ textContains: datasetName }))
+  await waitForNoSpinners(page)
+  await click(page, clickable({ textContains: 'Preview data' }))
+  await waitForNoSpinners(page)
+
+  const previewTableName = 'Cell Suspension Preview Data'
+  await findTableCellText(page, getTableHeaderPath(previewTableName, 1), 'cell_suspension_id')
+  await findTableCellText(page, getTableHeaderPath(previewTableName, 2), 'version')
+  await findTableCellText(page, getTableHeaderPath(previewTableName, 3), 'content')
+  await findTableCellText(page, getTableCellPath(previewTableName, 2, 1), 'f0caec4a-2a37-4895-8304-83d0fd0da588')
```

Comment:
If this data were to change, would it be appropriate for the test to fail, or does this make it a bit fragile?

Reply:
That's a good point! I wanted to confirm that the preview page looks as expected here, but if anyone deletes this snapshot the test would fail. Is it possible to mock data in this test?